code/game/objects/items/wayfinding.dm

```diff
@@ -155,57 +157,48 @@
 		user.put_in_hands(P)
 		P.owner = user.real_name
 
-/obj/machinery/pinpointer_dispenser/attackby(obj/item/I, mob/user, params)
+/obj/machinery/pinpointer_dispenser/attackby(obj/item/I, mob/living/user, params)
 	if(machine_stat & (BROKEN|NOPOWER))
 		return ..()
 
 	if(istype(I, /obj/item/pinpointer/wayfinding))
 		var/obj/item/pinpointer/wayfinding/WP = I
 
-		to_chat(user, "<span class='notice'>You put \the [WP] in the return slot.</span>")
-
-		var/refundiscredits = FALSE
 		var/itsmypinpointer = TRUE
-
-		//Will they meet the conditions to get a credit reward for recycling?
 		if(WP.owner != user.real_name)
 			itsmypinpointer = FALSE
 
-			if(synth_acc.has_money(refund_amt) && !WP.roundstart) //can it afford to refund and is the pinpointer not from the quirk
-				refundiscredits = TRUE
-				qdel(WP)
-				synth_acc._adjust_money(-refund_amt)
-				var/obj/item/holochip/holochip = new (loc)
-				holochip.credits = refund_amt
-				holochip.name = "[holochip.credits] credit holochip"
-				if(ishuman(user))
-					var/mob/living/carbon/human/customer = user
-					customer.put_in_hands(holochip)
-
-		if(!refundiscredits)
-			qdel(WP)
+		var/is_a_thing = "are [refund_amt] credit\s."
+		if(refund_amt > 0 && synth_acc.has_money(refund_amt) && !WP.roundstart)
+			synth_acc._adjust_money(-refund_amt)
+			var/obj/item/holochip/holochip = new (user.loc)
+			holochip.credits = refund_amt
+			holochip.name = "[holochip.credits] credit holochip"
+			if(ishuman(user))
+				var/mob/living/carbon/human/customer = user
+				customer.put_in_hands(holochip)
+		else if(!itsmypinpointer)
 			var/costume = pick(subtypesof(/obj/effect/spawner/bundle/costume))
 			new costume(user.loc)
-
-		set_expression("veryhappy", 2 SECONDS)
-
-		var/is_a_thing = "are [refund_amt] credit\s."
-		if(!refundiscredits)
 			is_a_thing = "is a freshly synthesised costume!"
 			if(prob(funnyprob))
 				is_a_thing = "is a pulse rifle! Just kidding it's a costume."
```

Comment:
This text is now outdated.

Reply:
I don't understand what you mean by that.

Reply:
Nevermind, my mistake--didn't see costumes were still occasionally being dispensed.